src/java/org/apache/cassandra/db/marshal/AbstractType.java

```diff
@@ -509,6 +538,41 @@ public boolean referencesDuration()
         return AssignmentTestable.TestResult.NOT_ASSIGNABLE;
     }
 
+    /**
+     * Produce a byte-comparable representation of the given value, i.e. a sequence of bytes that compares the same way
+     * using lexicographical unsigned byte comparison as the original value using the type's comparator.
+     *
+     * We use a slightly stronger requirement to be able to use the types in tuples. Precisely, for any pair x, y of
+     * non-equal valid values of this type and any bytes b1, b2 between 0x10 and 0xEF,
+     * (+ stands for concatenation)
+     *   compare(x, y) == compareLexicographicallyUnsigned(asByteComparable(x)+b1, asByteComparable(y)+b2)
+     * (i.e. the values compare like the original type, and an added 0x10-0xEF byte at the end does not change that) and:
+     *   asByteComparable(x)+b1 is not a prefix of asByteComparable(y)      (weakly prefix free)
+     * (i.e. a valid representation of a value may be a prefix of another valid representation of a value only if the
+     * following byte in the latter is smaller than 0x10 or larger than 0xEF). These properties are trivially true if
+     * the encoding compares correctly and is prefix free, but also permits a little more freedom that enables somewhat
+     * more efficient encoding of arbitrary-length byte-comparable blobs.
+     *
+     * Depending on the type, this method can be called for null or empty input, in which case the output is allowed to
+     * be null (the clustering/tuple encoding will accept and handle it).
+     *
+     * See the design document at
+     * https://docs.google.com/document/d/1X9S69gh2-qkb5DyWOFtu-DAr4tBKtzmGJP-Glr8uKL8/edit#heading=h.6dsgurnp1rij
```

Comment:
This document references DB-111. Are we happy with this? or do we want to change the document to remove references to Datastax specific tickets?